common/views/components/RenderlessPaginator/RenderlessPaginator.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+import { Fragment, type Node as ReactNode } from 'react';
+
+type Link = {|
+  +pathname: string,
+  +query: Object,
+|};
+
+type LinkProps = {|
+  href: Link,
+  as: Link,
+|};
+
+type Props = {|
+  currentPage: number,
+  pageSize: number,
+  totalResults: number,
+  link: LinkProps,
+  children: ({|
+    currentPage: number,
+    totalPages: number,
+    prevLink: any, // FIXME: ?Link
+    nextLink: any, // FIXME: ?Link
```

Comment:
[NextLinkType](https://github.com/wellcometrust/wellcomecollection.org/blob/master/common/model/next-link-type.js)

Reply:
Are these going to be optional?

Reply:
again, copy paste – I believe they should be optional so that when you're at the end of a list you don't have one or the other, yep.